icu4c/source/common/locid.cpp

```diff
@@ -778,6 +994,24 @@ Locale::minimizeSubtags(UErrorCode& status) {
     }
 }
 
+void
+Locale::canonicalize(UErrorCode& status) {
+    if (isBogus()) {
+        status = U_ILLEGAL_ARGUMENT_ERROR;
```

Comment:
Now you might clobber an incoming error code...
Please do either simply
```
Locale::canonicalize(UErrorCode& status) {
    if (U_FAILURE(status) || isBogus()) {
        return;
    }
```
or else
```
Locale::canonicalize(UErrorCode& status) {
    if (U_FAILURE(status)) {
        return;
    }
    if (isBogus()) {
        status = U_ILLEGAL_ARGUMENT_ERROR;
        return;
    }
```
I am not sure that it's necessary to set an error for isBogus() -- the simpler version simply does not do anything if "this" is bogus.

Reply:
done